espn_api/base_league.py

```diff
@@ -48,12 +50,15 @@ def _fetch_teams(self, data, TeamClass, pro_schedule = None):
         seasonId = data['seasonId']
 
         team_roster = {}
+        team_owners = {}
         for team in data['teams']:
             team_roster[team['id']] = team.get('roster', {})
+            team_owners[team['id']] = team.get('owners', [])
 
         for team in teams:
             roster = team_roster[team['id']]
-            self.teams.append(TeamClass(team, roster=roster, schedule=schedule, year=seasonId, pro_schedule=pro_schedule))
+            owners = team_owners[team['id']]

```

Comment:
You know what I just realized is the `team` data passing into the team class already has the owners data so we shouldn't need any of this logic in base_league. See comment in team class below

Reply:
Im not sure why I added roster like this either when its also in the `team` data object. I will clean up that logic later